subgraph/core/src/KlerosCore.ts

```diff
@@ -127,6 +127,42 @@ export function handleNewPeriod(event: NewPeriod): void {
     dispute.currentRuling = currentRulingInfo.getRuling();
     dispute.overridden = currentRulingInfo.getOverridden();
     dispute.tied = currentRulingInfo.getTied();
+
+    const rounds = dispute.rounds.load();
+    for (let i = 0; i < rounds.length; i++) {
+      const round = Round.load(rounds[i].id);
+      if (!round) continue;
+
+      const draws = round.drawnJurors.load();
+      // Iterate over all draws in the round
+      for (let j = 0; j < draws.length; j++) {
+        const draw = Draw.load(draws[j].id);
+        if (!draw) continue;
+
+        // Since this is a ClassicVote entity, this will only work for the Classic DisputeKit (which has ID "1").
+        const vote = ClassicVote.load(`${round.disputeKit}-${draw.id}`);
+
+        if (!vote) continue;
+
+        const juror = ensureUser(draw.juror);
+        juror.totalResolvedVotes = juror.totalResolvedVotes.plus(ONE);
+
+        if (vote.choice === null) continue;
+
+        // Check if the vote choice matches the final ruling
+        if (vote.choice!.equals(dispute.currentRuling)) {
+          juror.totalCoherentVotes = juror.totalCoherentVotes.plus(ONE);
+        }
+
+        // Recalculate coherenceScore
+        if (juror.totalResolvedVotes.gt(ZERO)) {
+          const coherenceScore = juror.totalCoherentVotes.times(BigInt.fromI32(100)).div(juror.totalResolvedVotes);
```

Comment:
it's missing the smoothing factor, check [here](https://github.com/kleros/kleros-v2/pull/1702/files#diff-e01d09c5f53e6045158d43b591c1868293037842fa12534f31d5433f4bf8bdcaL8)